.github/workflows/cicd.yaml

```diff
@@ -53,6 +53,39 @@ jobs:
         with:
           path: dist/*.whl
 
+  # Run tests against pydantic v1 until we drop support.
+  test-pydantic-v1:
```

Comment:
It was easier to just duplicate with a separate job so we can:
- remove quickly after dropping support for pydantic v1
- skip the build+upload steps (which would otherwise write duplicate wheels or require some more complex conditional logic)
- test the `poetry` locked version in the main job w/o a separate job to look it up first (or gross logic in "Install dependencies")